### PR TITLE
Add CI job to build Dockerfiles

### PR DESCRIPTION
**Pull request by aaronwolen**
Updates the CI to build and (selectively) push the docker images.

- images are always built when CI runs and the build step succeeds
- images are only pushed on master or with a new git tag

### CODE REVIEW

**Comment by Shelnutt2** [comment]
@aaronwolen can you remove the `.gitlab-ci.yaml` file since this will replace it?

File: README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # MyTile
 
-MariaDB storage engine based on [TileDB](https://tiledb.io/)
+[![Build Status](https://img.shields.io/azure-devops/build/tiledb-inc/836549eb-f74a-4986-a18f-7fbba6bbb5f0/8?label=Azure%20Pipelines&logo=azure-pipelines&style=flat-square)](https://dev.azure.com/TileDB-Inc/CI/_build/latest?definitionId=10&branchName=master)
+[![Docker-CLI](https://img.shields.io/static/v1?label=Docker&message=tiledb-mariadb&color=099cec&logo=docker&style=flat-square)](https://hub.docker.com/repository/docker/tiledb/tiledb-mariadb)
```
Comment (Shelnutt2) [inline README.md:4]: Wrong text, Perhaps something like "MariaDB CLI" and "MariaDB Server" ?

**Comment by aaronwolen** [comment]
I see that `[skip ci]` doesn't work w/ azure...

File: azure-pipelines.yml [hunk 2]
```diff
@@ -21,3 +21,24 @@ jobs:
     vmImage: $(imageName)
   steps:
   - template: scripts/azure-linux_mac.yml
+
+- job: Docker
```
Comment (Shelnutt2) [inline azure-pipelines.yml:25]: Should we make this into stages?

Comment (aaronwolen) [reply azure-pipelines.yml:25]: I haven't experimented with stages so I can't speak to the benefits/drawbacks.

**Pull request merged**